### ISSUE TITLE
`StreamingSieve::prime_pi` outputs incorrect results

### ISSUE THREAD

**Issue by aedoq** [huonw/primal#48]
Consider
```rust
fn main() {
    dbg!(primal::StreamingSieve::prime_pi(48));
    dbg!(primal::StreamingSieve::prime_pi(49));
    dbg!(primal::Sieve::new(49).prime_pi(49));
}
```
with output
```
[src/main.rs:7] StreamingSieve::prime_pi(48) = 15
[src/main.rs:8] StreamingSieve::prime_pi(49) = 16
[src/main.rs:9] primal::Sieve::new(49).prime_pi(49) = 15
```
The correct value for `pi(49)` is of course `15`. The value of `pi(50)` is calculated correctly again. Other problematic inputs I found are 121, 289, 961, 11047, 32611, 230907, 455166135

**Comment by cuviper** [comment]
Yikes! `git bisect` tells me that this regressed all the way back in commit 898c73e60bd07d857c15f62aea7046966cab9493.

**Comment by cuviper** [comment]
I think it was this change, from continuing `while s * s <= high` to breaking at `p * p >= high` -- should be `>`?

https://github.com/huonw/primal/commit/898c73e60bd07d857c15f62aea7046966cab9493#diff-856938cabe49ccdcbc3b559c45e4aff0dbb235921b5639a73cc273473ec0e760L191-R194

**Comment by cuviper** [comment]
See #49.

> Other problematic inputs I found are 121, 289, 961, 11047, 32611, 230907, 455166135

I can confirm that 121, 289, and 961 were broken, but the others were correct AFAICS. I did find other square primes that failed though, which makes sense for the bug that I found. Can you clarify how you were testing your inputs?

**Comment by aedoq** [comment]
> I can confirm that 121, 289, and 961 were broken, but the others were correct AFAICS. I did find other square primes that failed though, which makes sense for the bug that I found. Can you clarify how you were testing your inputs?

Well, I compared the values with the output from `num_prime`'s implementation and... it turns out [that one is broken too](https://github.com/cmpute/num-prime/issues/6).

**Comment by cuviper** [comment]
The fix is published in primal-sieve 0.3.4. Thanks for reporting the problem!
